buildSrc/src/main/java/org/elasticsearch/gradle/info/GlobalBuildInfoPlugin.java

```diff
@@ -248,7 +250,20 @@ public static String gitRevision(File rootDir) {
             }
             final String ref = readFirstLine(head);
             if (ref.startsWith("ref:")) {
-                revision = readFirstLine(gitDir.resolve(ref.substring("ref:".length()).trim()));
+                String branchname = ref.substring("ref:".length()).trim();
+                Path refFile = gitDir.resolve(branchname);
+                if (Files.exists(refFile)) {
+                    revision = readFirstLine(refFile);
+                } else {
+                    // Check packed references for commit ID
+                    Pattern p = Pattern.compile("^([a-f0-9]+) " + branchname + "$");
+                    revision = Files.lines(dotGit.resolve("packed-refs"))
```

Comment:
We should create this stream in a try-with-resources block to ensure we properly close it. In fact, we should refactor `readFirstLine()` below to do the same here as the Gradle daemon could be unintentionally holding a bunch of file handles here.

```
try  (Stream<String> lines = Files.lines(dotGit.resolve("packed-refs")) {
    // do stuff
}
```

Doc of `Files.lines()`:
```
     * This method must be used within a try-with-resources statement or similar
     * control structure to ensure that the stream's open file is closed promptly
     * after the stream's operations have completed.
```